### PR TITLE
Laboratory work #2, Maria Manukhina - 23FPL1

### PR DESCRIPTION
**Pull request by MariaManukhina**

### CODE REVIEW

**Comment by Vasilisa-Blyudova** [comment]
Unfortunately, I don't see any progress in your PR. It's time to start implementing Laboratory work #2.

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -18,6 +18,20 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    my_text = text.lower()
+    my_text.split()
```
Comment (Vasilisa-Blyudova) [inline lab_2_retrieval_w_bm25/main.py:24]: there is already `split` in the loop on line 26

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -18,6 +18,20 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    my_text = text.lower()
```
Comment (Vasilisa-Blyudova) [inline lab_2_retrieval_w_bm25/main.py:23]: try to use `lower` in loop before `split`

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Good work! Try to implement the laboratory work on 6

**Comment by MariaManukhina** [comment]
If I add right formula, which is in file, pycharm says unexpected character after line continuation character. If I changed \ to /, it says invalid syntax.

File: .github/workflows/ci_pipeline.yml [hunk 3]
```diff
@@ -85,7 +85,7 @@ jobs:
   unittests:
     name: Unittests
     runs-on: ubuntu-latest
-    timeout-minutes: 7
+    timeout-minutes: 11
```
Comment (Vasilisa-Blyudova) [inline .github/workflows/ci_pipeline.yml:88]: revert

Comment (artyomtugaryov) [reply .github/workflows/ci_pipeline.yml:88]: Amazing move! Agree with Vasilisa Mikhailovna

File: lab_2_retrieval_w_bm25/main.py [hunk 4]
```diff
@@ -22,13 +21,13 @@ def tokenize(text: str) -> list[str] | None:
...
-    return text.lower().split()
-
-
+    tokenized_text_clear = []
+    for word in text:
+        if not word.isalpha():
+            text.replace(word, " ")
+            tokenized_text_clear.append(" ")
+    tokenized_text = " ".join(tokenized_text_clear).split()
+    return tokenized_text
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:30]: add a new empty line after the function

File: lab_2_retrieval_w_bm25/main.py [hunk 5]
```diff
@@ -91,18 +83,15 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
...
-            not all(isinstance(token, str) for token in document_tokens) or not document_tokens:
+    if not isinstance(vocab, list[str], document_tokens, list[str]):
         return None
-
-    result = {}
-    for word in set(vocab) | set(document_tokens):
-        result[word] = document_tokens.count(word) / len(document_tokens)
-    return result
-
+    n_t = document_tokens.count
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:88]: I do not believe that it works

File: lab_2_retrieval_w_bm25/main.py [hunk 6]
```diff
@@ -148,17 +131,10 @@ def calculate_tf_idf(tf: dict[str, float], idf: dict[str, float]) -> dict[str, f
...
-    if not idf or not isinstance(idf, dict) or not all(isinstance(key, str) for key in idf) or \
-            not all(isinstance(value, float) for value in idf.values()):
+    if not isinstance(tf, idf, dict[str,float]):
         return None
-
-    tf_idf = {}
-    for word in tf:
-        tf_idf[word] = tf[word] * idf[word]
-    return tf_idf
+    calc_tf_idf = tf / idf
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:136]: so not create the variable

File: lab_2_retrieval_w_bm25/start.py [hunk 7]
```diff
@@ -25,127 +21,9 @@ def main() -> None:
...
-    spearman_bm25_bm25_cutoff = calculate_spearman(bm25_cutoff_ranks_only, bm25_ranks_only)
-    print('spearman for tf-idf and bm25:', spearman_tf_idf_bm25)
-    print('spearman for tf-idf and bm25 with cutoff:', spearman_tf_idf_bm25_cutoff)
-    print('spearman for bm25 and bm25 with cutoff:', spearman_bm25_bm25_cutoff)
-
-    result = bm25_cutoff_ranks_only
-    print('golden standard:', result)
+#    result = None
     assert result, "Result is None"
+    return None
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/start.py:26]: remove

File: lab_2_retrieval_w_bm25/start.py [hunk 8]
```diff
@@ -25,127 +21,9 @@ def main() -> None:
...
-    spearman_tf_idf_bm25 = calculate_spearman(tf_idf_ranks_only, bm25_ranks_only)
-    spearman_tf_idf_bm25_cutoff = calculate_spearman(tf_idf_ranks_only, bm25_cutoff_ranks_only)
-    spearman_bm25_bm25_cutoff = calculate_spearman(bm25_cutoff_ranks_only, bm25_ranks_only)
-    print('spearman for tf-idf and bm25:', spearman_tf_idf_bm25)
-    print('spearman for tf-idf and bm25 with cutoff:', spearman_tf_idf_bm25_cutoff)
-    print('spearman for bm25 and bm25 with cutoff:', spearman_bm25_bm25_cutoff)
-
-    result = bm25_cutoff_ranks_only
-    print('golden standard:', result)
+#    result = None
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/start.py:24]: looks like it will rise AssertionError

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**